Source/WebCore/workers/service/ServiceWorkerContainer.cpp

```diff
@@ -600,6 +605,19 @@ SWClientConnection& ServiceWorkerContainer::ensureSWClientConnection()
     return *m_swConnection;
 }
 
+Ref<SWClientConnection> ServiceWorkerContainer::ensureProtectedSWClientConnection()
+{
+    ASSERT(scriptExecutionContext());
+    if (!m_swConnection || m_swConnection->isClosed()) {
+        // Using RefPtr here results in an m_adoptionIsRequired assert.
```

Comment:
We should call `relaxAdoptionRequirement();` in the constructor of the object that's triggering this. We want to be using assert pointers.

also, this function should just call `ensureSWClientConnection()`, not duplicate it..